### ISSUE TITLE
errorStatus and Status are not comparable

### ISSUE THREAD

**Issue by mohibeyki** [oakserver/oak#89]
The example for ErrorHandling in the readme does not work, stating that `Type 'Status.NotFound' is not comparable to type 'ErrorStatus'.`

This is the said part of the readme
```
app.use(async (ctx, next) => {
  try {
    await next();
  } catch (err) {
    if (isHttpError(err)) {
      switch (err.status) {
        case Status.NotFound:
          // handle NotFound
          break;
        default:
        // handle other statuses
      }
    } else {
      // rethrow if you can't handle the error
      throw err;
    }
  }
});
```

The fix that I could come up with is not pretty and is to cast `err.status as Status` but that is only possible by casting into `unknown` first and then into `Status`
```
switch (err.status as unknown as Status) {
```

Am I missing something?

**Comment by gjuoun** [comment]
I think a good solution if you use ``` ErrorStatus.NotFound``` instead of ```Status.NotFound```

Of course, don't forget
```ts
import { ErrorStatus } from "https://deno.land/x/oak/types.ts";
```

**Comment by mohibeyki** [comment]
> I think a good solution if you use ` ErrorStatus.NotFound` instead of `Status.NotFound`
> 
> Of course, don't forget
> 
> ```ts
> import { ErrorStatus } from "https://deno.land/x/oak/types.ts";
> ```

My bad, I did not try to import it for some reason, but the problem persists on the readme file.
Also, I don't get why they are not comparable, Status is a superset of ErrorStatus, why is this happening?

**Comment by gjuoun** [comment]
If you are checking if it's HTTP error, ```err``` is actually a type of ```HttpError```

However, ```HttpError.status``` is a type of ```ErrorStatus```
```ts
export class HttpError extends Error {
  expose = false;
  status = ErrorStatus.InternalServerError;
}
```

**Comment by kitsonk** [comment]
I should make this clearer in the documentation

**Comment by kitsonk** [comment]
Actually, upon reflection, this is a bug.  I need to restructure how the status' work.

**Comment by kitsonk** [comment]
Ok, I think I fixed it in 47e88ac01f7055b5613589566283205a21e13d8a.  Comparing `err.status` with `Status` should work.  `ErrorStatus` is not a seperate enum anymore, just a type and there is the type guard `isErrorStatus(status)` which will help narrow the types.
